### PR TITLE
Added UserStatusChangeProcessor

### PR DESCRIPTION
**Pull request by delvh**
* Fixed the issue regarding the pending messages at login.
* Implemented message status change mechanism on the server side.
* Added `UserStatusChangeProcessor`.

### CODE REVIEW

File: src/main/java/envoy/server/database/PersistenceManager.java [hunk 1]
```diff
@@ -108,6 +108,61 @@ public void updateConfigItem(ConfigItem configItem) {
...
+	 * @param action the action that should be applied for this entity. Case does
+	 *               <b>not</b> matter.<br>
+	 *               Currently supported values are:<br>
+	 *               -"add"/"persist" to add an object to the database<br>
+	 *               -"update"/"merge" to update this object in the database<br>
+	 *               -"delete"/"remove" to delete this object in the database
+	 * @param entity the object to apply the action to
+	 * @since Envoy Server Standalone v0.1-alpha
+	 */
+	public void applyAction(String action, Object entity) {
```
Comment (CyB3RC0nN0R) [inline src/main/java/envoy/server/database/PersistenceManager.java]: I think this is not necessary.

File: src/main/java/envoy/server/processors/UserStatusChangeProcessor.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	 */
+	private static void notifyContacts(User user) {
+		UserStatusChangeEvent	evt					= new UserStatusChangeEvent(user.getId(), user.getStatus());
+		ConnectionManager		connectionManager	= ConnectionManager.getInstance();
+		try {
+			for (User contact : user.getContacts())
+				if (connectionManager.isOnline(contact.getId())) writeProxy.write(connectionManager.getSocketId(contact.getId()), evt);
+		} catch (IOException e) {
+			e.printStackTrace();
+			System.err.println("Could not notify online contacts of user " + evt.getId() + " that his status changed");
```
Comment (DieGurke) [inline src/main/java/envoy/server/processors/UserStatusChangeProcessor.java]: insert a "has" before the "changed".

Comment (CyB3RC0nN0R) [reply src/main/java/envoy/server/processors/UserStatusChangeProcessor.java]: This just makes it worse.

Comment (DieGurke) [reply src/main/java/envoy/server/processors/UserStatusChangeProcessor.java]: No it doesnt

Comment (DieGurke) [reply src/main/java/envoy/server/processors/UserStatusChangeProcessor.java]: Your oral request was event more dumb XD

Comment (CyB3RC0nN0R) [reply src/main/java/envoy/server/processors/UserStatusChangeProcessor.java]: I suggest "has been".

**Pull request merged**